### ISSUE TITLE
Inside amsmath alignment, the tcblayer counter gets an invalid value 

### ISSUE THREAD

**Issue by blefloch** [T-F-S/tcolorbox#160]
Using tcbox inside an amsmath alignment environment fails with a `Missing number` error, because the tcbox code tries to use the box `tcb@footnote@\romannumeral\c@tcblayer`, which only exists for positive `\c@tcblayer`.  In an alignment, `\c@tcblayer` ends up being zero because `\stepcounter` and `\addtocounter` have no effect during the repeated typesettings used by amsmath alignments.  Replacing them by explicit `\advance` (uncomment the `\patchcmd` lines) fixes the problem, with no adverse effect since tcbox is already careful to correctly restore this particular counter.
```
\documentclass{article}
\usepackage{tcolorbox,amsmath}
\makeatletter
% \patchcmd \tcb@layer@inc {\stepcounter{tcblayer}} {\global\advance\c@tcblayer 1 }{}{}
% \patchcmd \tcb@layer@dec {\addtocounter{tcblayer}{-1}} {\global\advance\c@tcblayer-1 }{}{}
\makeatother
\begin{document}
\begin{align}
  \text{\tcbox{abc}}
\end{align}
\end{document}
```

**Comment by T-F-S** [comment]
This is a really unexpected problem (for me). Thank you for reporting this.

I will change the layer counter settings from LaTeX to TeX code as you have suggested.

**Comment by T-F-S** [comment]
Fixed with version `5.0.0 (2021-12-16)`.
